enterprise/tailnet/pgcoord_test.go

```diff
@@ -871,51 +864,50 @@ func TestPGCoordinator_Lost(t *testing.T) {
 	agpltest.LostTest(ctx, t, coordinator)
 }
 
-func TestPGCoordinator_DeleteOnClose(t *testing.T) {
+func TestPGCoordinator_NoDeleteOnClose(t *testing.T) {
```

Comment:
I would also like to see tests specifically around coordinators closing and peers reconnecting.  These can use the `tailnet/test` `Peer` client, although you will likely have to do a little dev in that framework to allow us to assert that we _don't_ get certain updates.

First test:

1. 2 coordinators (A,B) with peers (1,2) connected as 1 -> A, 2 -> B.
1. Peer 1 adds a Tunnel to Peer 2.
1. Both Peers send node updates, assert we get them.
1. Close Coordinator A.
1. Assert that Peer 1 loses its coordination connection, and then have it connect to B, and re-establish a tunnel
1. Assert that Peer 1 gets Peer 2's Node.
1. Peer 1 sends a node with a different Preferred DERP than before. Assert that Peer 2 gets the new PreferredDERP _and_ that it never got a DISCONNECT for Peer 1 (although it _may or may not_ get a "LOST" update).

Second test: same as the first, but heartbeats from Coordinator A time out before Peer 1 reconnects.